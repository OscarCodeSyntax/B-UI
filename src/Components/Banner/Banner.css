
.margin-login-icon {
    margin: 20px 20px 20px 0px;
}

.margin-login-text {
    margin: 20px 20px 5px 20px;
}