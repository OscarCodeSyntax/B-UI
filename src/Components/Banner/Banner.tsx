import React from "react";
import "../../App.css";
import "./Banner.css";
import LoginLogoutIcon from "../LoginLogoutIcon/LoginLogoutIcon";

function Banner() {
  const isLoggedIn = localStorage.getItem("isLoggedIn");

  return (
    <div className="banner flex">
      <div>
        <div className="font-size-50 super-bold-black">B//APP</div>
        <div>Basic UI and DataService</div>
      </div>

      {/* split the content apart on this from left to right */}

      {LoginLogoutIcon(isLoggedIn)}
    </div>
  );
}

export default Banner;
