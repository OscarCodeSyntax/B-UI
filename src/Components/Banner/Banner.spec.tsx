import React from "react";
import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react";
import Banner from "./Banner";

describe("Banner ", () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  it("renders correctly", () => {
    const BannerWrapper = renderer.create(<Banner />).toJSON();
    expect(BannerWrapper).toMatchSnapshot();
  });

  it("renders 4 div element", () => {
    const testRenderer = renderer.create(<Banner />);
    const testInstance = testRenderer.root;

    expect(testInstance.findAllByType("div").length).toEqual(7);
  });

  it("has all the correct text", () => {
    render(<Banner />);
    expect(screen.getByText("B//APP")).toBeTruthy();
    expect(screen.getByText("Basic UI and DataService")).toBeTruthy();
  });
});
