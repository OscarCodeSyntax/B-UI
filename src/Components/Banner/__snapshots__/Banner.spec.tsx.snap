// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner  renders correctly 1`] = `
<div
  className="banner flex"
>
  <div>
    <div
      className="font-size-50 super-bold-black"
    >
      B//APP
    </div>
    <div>
      Basic UI and DataService
    </div>
  </div>
  <div
    className="split padding-20"
  >
    <a
      className="flex color-black"
      href="/"
      onClick={[Function]}
    >
      <div>
        <div
          className="margin-5"
        >
           Login
        </div>
      </div>
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge margin-login  css-tzssek-MuiSvgIcon-root"
        data-testid="LoginIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M11 7 9.6 8.4l2.6 2.6H2v2h10.2l-2.6 2.6L11 17l5-5zm9 12h-8v2h8c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-8v2h8z"
        />
      </svg>
    </a>
  </div>
</div>
`;
